### ISSUE TITLE
XMRigCCServer: HTTP Daemon failed to start when "use-tls" : true in config_cc.json

### ISSUE THREAD

**Issue by adi3yz** [Bendr0id/xmrigCC#103]
i use this
Option 1 - Generate TLS certificates
openssl genrsa -out server.key 1024
openssl req -days 365 -out server.pem -new -x509 -key server.key
copy both file to xmrigcc folder but when i run ccserver
HTTP Daemon failed to start.

**Comment by ariadarkkkis** [comment]
I have same problem on windows.I used Gobetween (http://gobetween.io/downloads.html) to manage tls and redirect to cc without tls option enabled.

**Comment by Bendr0id** [comment]
> btw how to change ccserver time interval check client? bcoz i set client update to cc 180s
and now dashboard keep showing client offline

In index.html file you'll find there vars. Just modify them for your needings. The TRESHOLD is the one you should increase. Maybe 240 or 300 * 1000 is a good candidate for you.
 
```
    var TRESHOLD_IN_MS = 60 * 1000; 
    var RELOAD_INTERVAL_IN_MS = 10 * 1000;
```

**Comment by Bendr0id** [comment]
On the SSL problems i started investigation. Currently its unclear what is goind wrong on Windows. But it seems that even the sample app from libmicrohttpd with tls isnt working on Windows.

**Comment by lost-bro** [comment]
Good day Bendr0id
Just want to confirm that I am experiencing the same issue:  The TLS option (windows 7 box) for CCserver is also giving the "HTTP Daemon failed to start." error.  I compiled from code and get the same results.
Is there a work-around for this issue?  I finally managed to get Xmrig-proxy interfaced w/ a front-end TLS  
proxy to have all miner > proxy communication encrypted.  Would be super nice to also get the CCServer > miner communications encrypted.
Thanks in advance
take care, peace
lost_bro

**Comment by Bendr0id** [comment]
I finally found the root cause. It's due to libmicrohttpd isn't compiled with tls support. Either i find a version which has it build in or i need to compile it on my own..

Any help appreciated

**Comment by mk148a** [comment]
> libmicrohttpd

hi, do you have any update? This issue is continue for me :S

**Comment by Bendr0id** [comment]
Or just host xmrigCCServer on a linux system.

**Comment by mk148a** [comment]
> 
> 
> Or just host xmrigCCServer on a linux system.

i am trying on linux but same result, can you help for step by step?

**Comment by rASSSOL** [comment]
Hi! I Have a same problem.
I use this:
Option 1 - Generate TLS certificates
openssl genrsa -out server.key 1024
openssl req -days 365 -out server.pem -new -x509 -key server.key
copy both file to CCserver folder but when i run CCserver it's failed to start. But xmrigCCproxy works with tls.
Win10 x64, last versions xmrigCCproxy and xmrigCC. Termux Openssl 1.1.1b 26 Feb 2019.

**Comment by GjBrutello** [comment]
* VERSIONS:     XMRigCC/1.9.3 (based on XMRig) libuv/1.23.0 MSVC/2017 (RELEASE with TLS)
 * PUSHSERVICE:  Disabled
 * COMMANDS:     quit
[2019-05-26 20:17:00] HTTP Daemon failed to start.

No one version windows support the HTTPS Daemon. I want to encrypt traffic on the local network.

**Comment by Bendr0id** [comment]
The problem is that the http third party library which is used doesn't support windows tls. If you want to have it running with TLS use linux or wps with the Linux version.

**Comment by GjBrutello** [comment]
then it's weird but xmrig-proxy works just fine without any haproxy and Ubuntu. Maybe you could make a fork of its libs? Proof: [https://savepic.pro/image/NcGpj](url)
![xmrig_ssl](https://user-images.githubusercontent.com/50217444/58447799-70f01280-8127-11e9-9ec2-dd90bc0725ad.png)

**Comment by Bendr0id** [comment]
It's not wierd. XMRigCC minerwise is also able todo full TLS on windows. Problem is the http server part. Xmrig-proxy is using ssl for stratum only. (Connection proxy to pool and Miner to proxy) the API of xmrig-proxy doesn't support https. So it's exactly the same for cc server.

**Comment by Bendr0id** [comment]
XMRigCC server part is not a proxy!

**Comment by distributorofpain** [comment]
Thanks for putting this app together(works good).  Having the same issue with Windows 10, i see its basically not supported at this time.  Thought i was generating certificates wrong, so i kept trying different things until i did a quick search in the issues section(and found this).

Could someone possibly make a note on the wiki page(assuming a fix is not imminent)?

**Comment by Bendr0id** [comment]
Fixed in #2.1.0
